### PR TITLE
Initial TravisCI configuration.

### PR DESCRIPTION
**Pull request by jmchilton**
This does two things - runs the Python unit tests and lints a subset of the Python code. I wanted to also run the QUnit tests but they are not currently passing in my testing.

Example Report: https://travis-ci.org/jmchilton/galaxy/builds/55743614

### CODE REVIEW

**Comment by dannon** [comment]
This is an automated message. Thanks for your contribution, a [Trello card](https://trello.com/c/yUIi1pLx) to track this issue has been created. [Apply this patch](https://github.com/galaxyproject/galaxy/pull/40.patch) for testing.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+# 201 and 202 are spaces after ( and before ), 501 is line length. These
+# are exceptions allowed (encouraged?) by Galaxy style guidelines.
```
Comment (hexylena) [inline setup.cfg:3]: Innocent question: why _are_ these encouraged? 201/2 not 501, 501 is more understandable.

Comment (martenson) [inline setup.cfg:3]: Galaxy coding style requires (spaces [ { in } brackets ] ) for readability. It is team-arranged standard.

**Comment by martenson** [comment]
This is great John! :100:

File: setup.cfg (hunk 1 above)
Comment (hexylena) [inline setup.cfg:3]: Huh, ok. Could there be documentation on all of the conventions? (Maybe in Contributing.md?)

Comment (martenson) [inline setup.cfg:3]: https://wiki.galaxyproject.org/Develop/BestPractices
Adding to GH is a good idea. I will bounce it around.

Comment (jmchilton) [inline setup.cfg:3]: "requires" feels like a strong word here - I have interpreted the language as being maybe something more like "recommends" or "suggests"? Some people don't include the whitespace - and I respect their civil disobedience. 

From Wiki:

"Whitespace around operators and inside parentheses is generally included, particularly if it helps readability."

Comment (erasche) [inline setup.cfg]: (@jmchilton you may have accidentally a word there which removes some of the meaning of the sentence: "...their their civil disobediance". Feel free to remove this comment if/when you update)

Comment (jmchilton) [inline setup.cfg:3]: @erasche 

contributing.md explicitly mentions the style guidelines and links to them:

"If your changes modify code - please ensure the resulting files conform to Galaxy [style guidelines](https://wiki.galaxyproject.org/Develop/BestPractices)."

Comment (hexylena) [inline setup.cfg:3]: @jmchilton so it does, I'm blind. Skimmed too quickly.

Comment (martenson) [inline setup.cfg:3]: 'Requires' is indeed stronger word than needed. If you disobey I won't hunt you down either. :)

**Comment by nsoranzo** [comment]
:+1: Thanks John!

**Pull request merged**